### ISSUE TITLE
BYUtv 

### ISSUE THREAD

**Issue by oneharrydude** [ytdl-org/youtube-dl#30433]
## Checklist

- [x ] I'm reporting a broken site support issue
- [ ] I've verified that I'm running youtube-dl version **2021.12.17**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [ ] I've searched the bugtracker for similar bug reports including closed ones
- [ ] I've read bugs section in FAQ

## Verbose log

```
C:\Users\User>youtube-dl --cookies "cookies.txt" https://content-auso1.uplynk.com/preplay2/20cfbfc8748e4f3ca38259cf3bd427d5/a5610f606df0099a1a3c572feab7fd27/lMqLapmV9uQE1G1OjMIg46VDFeo9UlhkS0aI7CBBE0q.m3u8?pbs=e5cb15aa0ce9411fbe29fdd3c6b64342 --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--cookies', 'cookies.txt', 'https://content-auso1.uplynk.com/preplay2/20cfbfc8748e4f3ca38259cf3bd427d5/a5610f606df0099a1a3c572feab7fd27/lMqLapmV9uQE1G1OjMIg46VDFeo9UlhkS0aI7CBBE0q.m3u8?pbs=e5cb15aa0ce9411fbe29fdd3c6b64342', '--verbose']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.8.1 (CPython) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 2021-02-02-git-2367affc2c-essentials_build-www.gyan.dev, ffprobe 2021-02-02-git-2367affc2c-essentials_build-www.gyan.dev
[debug] Proxy map: {}
[generic] lMqLapmV9uQE1G1OjMIg46VDFeo9UlhkS0aI7CBBE0q: Requesting header
WARNING: Could not send HEAD request to https://content-auso1.uplynk.com/preplay2/20cfbfc8748e4f3ca38259cf3bd427d5/a5610f606df0099a1a3c572feab7fd27/lMqLapmV9uQE1G1OjMIg46VDFeo9UlhkS0aI7CBBE0q.m3u8?pbs=e5cb15aa0ce9411fbe29fdd3c6b64342: HTTP Error 403: Forbidden
[generic] lMqLapmV9uQE1G1OjMIg46VDFeo9UlhkS0aI7CBBE0q: Downloading webpage
ERROR: Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "c:\users\user\appdata\local\programs\python\python38\lib\site-packages\youtube_dl\extractor\common.py", line 634, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "c:\users\user\appdata\local\programs\python\python38\lib\site-packages\youtube_dl\YoutubeDL.py", line 2288, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "c:\users\user\appdata\local\programs\python\python38\lib\urllib\request.py", line 531, in open
    response = meth(req, response)
  File "c:\users\user\appdata\local\programs\python\python38\lib\urllib\request.py", line 640, in http_response
    response = self.parent.error(
  File "c:\users\user\appdata\local\programs\python\python38\lib\urllib\request.py", line 569, in error
    return self._call_chain(*args)
  File "c:\users\user\appdata\local\programs\python\python38\lib\urllib\request.py", line 502, in _call_chain
    result = func(*args)
  File "c:\users\user\appdata\local\programs\python\python38\lib\urllib\request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)

```

## Description

I have tried various versions of the video on byutv's page https://www.byutv.org/player/af9e59e1-dcbb-403c-9e3d-7d5b95c352de/studio-c-the-disappearance-of-jason-gray. I am getting the forbidden flag on each of them. This used to work.

**Comment by dirkf** [comment]
The formats are found:
```
$ youtube-dl -v -F  'https://www.byutv.org/player/af9e59e1-dcbb-403c-9e3d-7d5b95c352de/studio-c-the-disappearance-of-jason-gray'
[debug] System config: ['--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-F', 'https://www.byutv.org/player/af9e59e1-dcbb-403c-9e3d-7d5b95c352de/studio-c-the-disappearance-of-jason-gray']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.5.2 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[BYUtv] studio-c-the-disappearance-of-jason-gray: Downloading JSON metadata
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading m3u8 information
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading MPD manifest
WARNING: [BYUtv] Unknown MIME type application/mp4 in DASH manifest
WARNING: [BYUtv] Unknown MIME type application/mp4 in DASH manifest
[info] Available formats for af9e59e1-dcbb-403c-9e3d-7d5b95c352de:
format code  extension  resolution note
hls-215      mp4        256x144     215k , avc1.4d000c, 15.0fps, mp4a.40.5
hls-420      mp4        416x234     420k , avc1.4d000d, 30.0fps, mp4a.40.5
hls-759      mp4        640x360     759k , avc1.4d001e, 30.0fps, mp4a.40.5
hls-1708     mp4        960x540    1708k , avc1.4d001f, 30.0fps, mp4a.40.5
hls-3242     mp4        1280x720   3242k , avc1.4d001f, 30.0fps, mp4a.40.5
hls-4750     mp4        1280x720   4750k , avc1.64001f, 60.0fps, mp4a.40.5
hls-6301     mp4        1280x720   6301k , avc1.64001f, 60.0fps, mp4a.40.5 (best)
```
However the streams are protected by SAMPLE-AES which neither yt-dl nor _ffmpeg_ understand.
```
...
[hls @ 0x202ed00] SAMPLE-AES encryption is not supported yet
...
```

**Comment by oneharrydude** [comment]
Are there any workarounds or other programs I could use? Thanks for the help!

**Comment by dirkf** [comment]
If you are studying there and can't access the material in your browser, ask the BYU admin to provide it in a different form.

**Comment by Vangelis66** [comment]
**dirkf** wrote on Dec 28th 2021: 

> ... the streams are protected by **SAMPLE-AES** which **neither** yt-dl nor **_ffmpeg_** understand

```
...
[hls @ 0x202ed00] SAMPLE-AES encryption is not supported yet
...
```
Actually, the above is **no longer accurate** as far as `ffmpeg` is concerned, because support for (decrypting) **SAMPLE-AES encryption** has been added to it 3 1/2 months ago: 

[libavformat/hls: add support for decryption of HLS media segments encrypted using SAMPLE-AES encryption method](https://github.com/FFmpeg/FFmpeg/commit/ff958b38463f)

The above was code snapshot `FFmpeg-n4.5-dev-2685-git-20211013-gff958b3` and such support is **already present** inside the most recent stable release, `ffmpeg-n5.0` :smile: .

However, SAMPLE-AES is just an **encryption scheme** and, unlike **AES-128**, is **very seldom used on its own** for content/media protection (but such implementations _do_ exist, **few** as they may be); notoriously, SAMPLE-AES is **associated with Apple's FairPlay** full-blown DRM and, am afraid, this is where the OP gets stuck with... :disappointed: 

When `ffmpeg-n5.0+` is fed **AppleFP DRM**'ed HLS streams, the error generated is now: 
```
Unable to open key file skd://[some URI here]
```
because FFmpeg **doesn't support Apple FairPlay**, just SAMPLE-AES!

`yt-dlp` is more informative of the situation: 
```
yt-dlp -F "https://www.byutv.org/player/af9e59e1-dcbb-403c-9e3d-7d5b95c352de/studio-c-the-disappearance-of-jason-gray" => 

[BYUtv] studio-c-the-disappearance-of-jason-gray: Downloading JSON metadata
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading MPD manifest
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading m3u8 information
[info] Available formats for af9e59e1-dcbb-403c-9e3d-7d5b95c352de:
ID       EXT RESOLUTION FPS |   TBR PROTO  | VCODEC        VBR ACODEC    ABR
----------------------------------------------------------------------------
hls-215  mp4 256x144     15 |  215k m3u8_n | avc1.4d000c  215k mp4a.40.5  0k
hls-420  mp4 416x234     30 |  420k m3u8_n | avc1.4d000d  420k mp4a.40.5  0k
hls-759  mp4 640x360     30 |  759k m3u8_n | avc1.4d001e  759k mp4a.40.5  0k
hls-1708 mp4 960x540     30 | 1708k m3u8_n | avc1.4d001f 1708k mp4a.40.5  0k
hls-3242 mp4 1280x720    30 | 3242k m3u8_n | avc1.4d001f 3242k mp4a.40.5  0k
hls-4750 mp4 1280x720    60 | 4750k m3u8_n | avc1.64001f 4750k mp4a.40.5  0k
hls-6301 mp4 1280x720    60 | 6301k m3u8_n | avc1.64001f 6301k mp4a.40.5  0k

but...

yt-dlp -f hls-759 "https://www.byutv.org/player/af9e59e1-dcbb-403c-9e3d-7d5b95c352de/studio-c-the-disappearance-of-jason-gray" -N 4 -o "byutv-studio-c-S14E10-the-disappearance-of-jason-gray.mp4" => 

[BYUtv] studio-c-the-disappearance-of-jason-gray: Downloading JSON metadata
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading MPD manifest
[BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading m3u8 information
[info] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading 1 format(s): hls-759
[hlsnative] Downloading m3u8 manifest
ERROR: This video is DRM protected; Try selecting another format with --format or add --check-formats to automatically fallback to the next best format
```
To add insult to injury, 

> [BYUtv] af9e59e1-dcbb-403c-9e3d-7d5b95c352de: Downloading MPD manifest

the MPD manifest is populated exclusively with **cenc** DRM'ed streams (use ` -F --allow-u` with `yt-dlp` to see), thus the OP is stuck between a rock and a hard place (i.e. ONLY DRM'ed streams exist for the programme he wants to grab...); as is common knowledge already, DRM-circumvention is besides the scope of `yt-dl` (and forks) and won't be discussed here at all... :stuck_out_tongue_winking_eye:
